match-making/update_session_url.go

```diff
@@ -16,75 +13,20 @@ func (commonProtocol *CommonProtocol) updateSessionURL(err error, packet nex.Pac
 	}
 
 	connection := packet.Sender().(*nex.PRUDPConnection)
 	endpoint := connection.Endpoint().(*nex.PRUDPEndPoint)
-	server := endpoint.Server
 
 	// * Mario Kart 7 seems to set an empty strURL, so I assume this is what the method does?
+	originalHost := session.GameMatchmakeSession.Gathering.HostPID
 	session.GameMatchmakeSession.Gathering.HostPID = connection.PID().Copy().(*types.PID)
-	if session.GameMatchmakeSession.Gathering.Flags.PAND(match_making.GatheringFlags.DisconnectChangeOwner) != 0 {
-		originalOwner := session.GameMatchmakeSession.Gathering.OwnerPID
-		session.GameMatchmakeSession.Gathering.OwnerPID = connection.PID().Copy().(*types.PID)
 
-		category := notifications.NotificationCategories.OwnershipChanged
-		subtype := notifications.NotificationSubTypes.OwnershipChanged.None
-
-		oEvent := notifications_types.NewNotificationEvent()
-		oEvent.PIDSource = originalOwner.Copy().(*types.PID)
-		oEvent.Type = types.NewPrimitiveU32(notifications.BuildNotificationType(category, subtype))
-		oEvent.Param1 = idGathering.Copy().(*types.PrimitiveU32)
-		oEvent.Param2 = types.NewPrimitiveU32(connection.PID().LegacyValue()) // TODO - This assumes a legacy client. Will not work on the Switch
-
-		// TODO - StrParam doesn't have this value on some servers
-		// * https://github.com/kinnay/NintendoClients/issues/101
-		// * unixTime := time.Now()
-		// * oEvent.StrParam = strconv.FormatInt(unixTime.UnixMicro(), 10)
-
-		stream := nex.NewByteStreamOut(endpoint.LibraryVersions(), endpoint.ByteStreamSettings())
-
-		oEvent.WriteTo(stream)
-
-		rmcRequest := nex.NewRMCRequest(endpoint)
-		rmcRequest.ProtocolID = notifications.ProtocolID
-		rmcRequest.CallID = common_globals.CurrentMatchmakingCallID.Next()
-		rmcRequest.MethodID = notifications.MethodProcessNotificationEvent
-		rmcRequest.Parameters = stream.Bytes()
-
-		rmcRequestBytes := rmcRequest.Bytes()
-
-		common_globals.Sessions[idGathering.Value].ConnectionIDs.Each(func(_ int, connectionID uint32) bool {
-			target := endpoint.FindConnectionByID(connectionID)
-			if target == nil {
-				common_globals.Logger.Warning("Client not found")
-				return false
-			}
-
-			var messagePacket nex.PRUDPPacketInterface
-
-			if target.DefaultPRUDPVersion == 0 {
-				messagePacket, _ = nex.NewPRUDPPacketV0(server, target, nil)
-			} else {
-				messagePacket, _ = nex.NewPRUDPPacketV1(server, target, nil)
-			}
-
-			messagePacket.SetType(constants.DataPacket)
-			messagePacket.AddFlag(constants.PacketFlagNeedsAck)
-			messagePacket.AddFlag(constants.PacketFlagReliable)
-			messagePacket.SetSourceVirtualPortStreamType(target.StreamType)
-			messagePacket.SetSourceVirtualPortStreamID(endpoint.StreamID)
-			messagePacket.SetDestinationVirtualPortStreamType(target.StreamType)
-			messagePacket.SetDestinationVirtualPortStreamID(target.StreamID)
-			messagePacket.SetPayload(rmcRequestBytes)
-
-			server.Send(messagePacket)
-
-			return false
-		})
```

Comment:
While this may not be for changing the owner, @DaniElectra does not sending *anything* here seem correct to you? I scanned all the MK7 uploads we've gotten through user submissions and MK7 DOES use the "Host changed" notification type, which seems like it would be sent here?

Reply:
I wasn't aware that the server sends the "Host changed" notification, I had assumed that it was only for recent versions of NEX

Considering that this is the only method that the game uses for updating the host, yeah it should probably be added